tests/js/web/DropZoneComponentTest.js

```diff
@@ -0,0 +1,928 @@
+import DropZoneInstanceManager from '../../../js/DropZone/DropZoneInstanceManager';
+import DropZoneComponentUtils from '../../../js/DropZone/DropZoneComponentUtils';
+import DropZoneComponentValidation from '../../../js/DropZone/DropZoneComponentValidationManager';
+import DropZoneBodyClassManager from '../../../js/DropZone/DropZoneBodyClassManager';
+
+describe('DropZoneComponent', () => {
+    let $html;
+    let $fileInput;
+    let $dropZone;
+    let $browse;
+    let $info;
+    let dropZoneComponent;
+    let instanceManager;
+    let optionsManager;
+    let utils;
+    let validation;
+    let classManager;
+
+    beforeEach(() => {
+        $html = $(document.documentElement);
+        $fileInput = $('<input type="text" id="fileInput">').appendTo($html);
+        $dropZone = $('<div data-dropzone-id="0" class="dropzone"></div>').appendTo($html);
+        $info = $('<p class="info"></p>').appendTo($dropZone);
+        $browse = $('<span class="browse">browse</span>').appendTo($info);
+
+        instanceManager = sinon.createStubInstance(DropZoneInstanceManager);
+        optionsManager = sinon.createStubInstance(DropZoneOptionsManager);
+        utils = sinon.createStubInstance(DropZoneComponentUtils);
+        validation = sinon.createStubInstance(DropZoneComponentValidation);
+        classManager = sinon.createStubInstance(DropZoneBodyClassManager);
+
+        instanceManager.getInstance.returns({ node: $dropZone[0], info: $info[0] });
+        instanceManager.getFiles.returns(['file']);
+        optionsManager.getInstanceOptions.returns({
+            nodeClasses: {
+                wrapper: 'wrapper',
+                browse: 'browse',
+                close: 'close'
+            }
+        });
+        utils.createFileNode.returns('<div class="file"><a class="close" href="#"></a></div>');
+
+        dropZoneComponent = new DropZoneComponent(
+            $html,
+            '.dropzone',
+            instanceManager,
+            optionsManager,
+            utils,
+            validation,
+            classManager
+        );
+    });
+
+    afterEach(() => {
+        $html.html('');
+    });
+
+    describe('init()', () => {
+        let inputStub;
+        let browseStub;
+
+        beforeEach(() => {
+            $('<div class="dropzone test"></div>').appendTo($html);
+            $('<div class="dropzone test"></div>').appendTo($html);
+            $('<div class="dropzone test"></div>').appendTo($html);
+            $('<div class="dropzone test"></div>').appendTo($html);
+
+            inputStub = sinon.stub(dropZoneComponent, 'processInputNode');
+            browseStub = sinon.stub(dropZoneComponent, 'processBrowseNode');
+            instanceManager.getInstance.returns([
+                { id: 0 },
+                { id: 1, input: {}, options: {}, browse: {} }
+            ]);
+        });
+
+        afterEach(() => {
+            $html.find('.dropzone.test').remove();
```

Comment:
I don't _think_ this is needed, as you create a new `$html` for every test above, which should be creating a new `<html>` element and discarding the old one (and all its descendants).